src/main/java/de/tum/in/test/api/ast/asserting/UnwantedNodesAssert.java

```diff
@@ -0,0 +1,124 @@
+ */
+public class UnwantedNodesAssert extends AbstractAssert<UnwantedNodesAssert, Path> {
+
+	/**
+	 * The flag, determining whether the path points to a file or to a directory
+	 */
+	private final boolean assertDirectory;
+
+	/**
+	 * The language level on which the JavaParser shall operate
+	 */
+	private final LanguageLevel level;
+
+	/**
+	 * (Designated constructor) Non-statically creates an unwanted node assertion
+	 * object
+	 *
+	 * @param actual          The path from which on the Java-Files are checked
+	 * @param assertDirectory A flag, determining whether the path points to a file
+	 *                        or to a directory
+	 * @param level           The language level on which the JavaParser operates
+	 */
+	public UnwantedNodesAssert(Path actual, boolean assertDirectory, LanguageLevel level) {
+		super(actual, UnwantedNodesAssert.class);
+		this.assertDirectory = assertDirectory;
+		this.level = level;
+	}
+
+	/**
+	 * (Convenience constructor) Non-statically creates an unwanted node assertion
+	 * object (language level is automatically set to Java 11)
+	 *
+	 * @param actual          The path from which on the Java-Files are checked
+	 * @param assertDirectory A flag, determining whether the path points to a file
+	 *                        or to a directory
+	 */
+	public UnwantedNodesAssert(Path actual, boolean assertDirectory) {
+		this(actual, assertDirectory, LanguageLevel.JAVA_11);
+	}
+
+	/**
+	 * (Convenience constructor) Non-statically creates an unwanted node assertion
+	 * object (language level is automatically set to Java 11 and the file/directory
+	 * flag is automatically set to file)
+	 *
+	 * @param actual The path from which on the Java-Files are checked
+	 */
+	public UnwantedNodesAssert(Path actual) {
+		this(actual, false, LanguageLevel.JAVA_11);
+	}
+
+	/**
+	 * Statically creates an unwanted node assertion object (language level is
+	 * automatically set to Java 11 and the file/directory flag is automatically set
+	 * to file)
+	 *
+	 * @param actual The path from which on the Java-Files are checked
+	 * @return An unwanted node assertion object (for chaining)
+	 */
+	public static UnwantedNodesAssert assertThat(Path actual) {
+		return new UnwantedNodesAssert(actual);
+	}
+
+	/**
+	 * Determines the path to point to a File
+	 *
+	 * @return An unwanted node assertion object (for chaining)
+	 */
+	public UnwantedNodesAssert asFile() {
+		return new UnwantedNodesAssert(actual, false);
+	}
+
+	/**
+	 * Determines the path to point to a Directory
+	 *
+	 * @return An unwanted node assertion object (for chaining)
+	 */
+	public UnwantedNodesAssert asDirectory() {
+		return new UnwantedNodesAssert(actual, true);
+	}
+
+	/**
+	 * Determines the language level on which the JavaParser operate
+	 *
+	 * @param level The language level on which the JavaParser shall operate
+	 * @return An unwanted node assertion object (for chaining)
+	 */
+	public UnwantedNodesAssert withConfiguration(LanguageLevel level) {
```

Comment:
Calling `assertThat(x).asFile().withConfiguration(LanguageLevel.Java_17)` works as expected, but with `assertThat(x).withConfiguration(LanguageLevel.Java_17).asFile()` the Language-Level will get overwritten again.

Reply:
I would suggest not making the constructors public API and reducing their number. This also reduces the likelihood of such errors.

Reply:
[Implemented part of the new feedback](https://github.com/ls1intum/Ares/pull/248/commits/ead5d75f0814dbd3738c57b058de8759b980e46d):
- Removed convenience constructors
- Made designated constructor private
- Changed `UnwantedNodesAssert.asFile()` and `UnwantedNodesAssert.asDirectory()`, so that the language-level does not get overwritten.